indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -57,18 +58,36 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Random;
 
 public class RealtimeIndexTask extends AbstractTask
 {
   private static final EmittingLogger log = new EmittingLogger(RealtimeIndexTask.class);
+  private final static Random random = new Random();
 
   private static String makeTaskId(FireDepartment fireDepartment)
   {
-    return String.format(
-        "index_realtime_%s_%d_%s",
+    return makeTaskId(
         fireDepartment.getDataSchema().getDataSource(),
         fireDepartment.getTuningConfig().getShardSpec().getPartitionNum(),
-        new DateTime().toString()
+        new DateTime(),
+        random.nextInt()
+    );
+  }
+
+  static String makeTaskId(String dataSource, int partitionNumber, DateTime timestamp, int randomBits)
+  {
+    final StringBuilder suffix = new StringBuilder(8);
+    for(int i = 0; i < Ints.BYTES * 2; ++i) {
+      suffix.append((char)('a' + ((randomBits >>> (i * 4)) & 0x0F)));
+    }
+    return String.format(
+        "index_realtime_%s_%d_%s_%s",
```

Comment:
This is simply doing something as close to what tranquility is doing.
A uuid is overkill for just the suffix, since we have millisecond level timestamp already, and having the timestamp is always useful for debuggin.